actionpack/lib/action_dispatch/system_testing/driver.rb

```diff
@@ -32,7 +32,21 @@ def register
         end
 
         def register_selenium(app)
-          Capybara::Selenium::Driver.new(app, { browser: @browser }.merge(@options)).tap do |driver|
+          if @browser == :headless_chrome
+            register_selenium_headless_chrome(app)
+          else
+            Capybara::Selenium::Driver.new(app, { browser: @browser }.merge(@options)).tap do |driver|
+              driver.browser.manage.window.size = Selenium::WebDriver::Dimension.new(*@screen_size)
+            end
+          end
+        end
+
+        def register_selenium_headless_chrome(app)
+          browser_options = Selenium::WebDriver::Chrome::Options.new
+          browser_options.args << "--headless"
+          browser_options.args << "--disable-gpu"
+
+          Capybara::Selenium::Driver.new(app, { browser: :chrome, options: browser_options }.merge(@options)).tap do |driver|
```

Comment:
This feels repetitive, with the only reason we need an extra method is because we need to merge `options: browser_options`.

What do you think about something like this:

```ruby
def browser_options
  if @browser == :headless_chrome
    @options.merge(options: "--headless --disable-gpu")
  else
    @options
  end
end

def register_selenium(app)
  Capybara::Selenium::Driver.new(app, { browser: @browser }.merge(browser_options)).tap do |driver|
    driver.browser.manage.window.size = Selenium::WebDriver::Dimension.new(*@screen_size)
  end
end
```

Reply:
:+1: Since that can not be passed by options, I just changed it and updated!